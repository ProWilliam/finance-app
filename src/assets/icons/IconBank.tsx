import * as React from "react";
import Svg, { SvgProps, Path } from "react-native-svg"; 

const IconBank = (props: SvgProps) => (
  <Svg
    xmlns="http://www.w3.org/2000/svg"
    width={50}
    height={50}
    fill="none"
    aria-hidden="true"
    {...props}
    viewBox="0 0 24 24"
  >
    <Path
      stroke="currentColor"
      strokeLinecap="round"
      strokeWidth={2}
      d="M8 7V6a1 1 0 0 1 1-1h11a1 1 0 0 1 1 1v7a1 1 0 0 1-1 1h-1M3 18v-7a1 1 0 0 1 1-1h11a1 1 0 0 1 1 1v7a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1Zm8-3.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
    />
  </Svg>
)
export default IconBank;
