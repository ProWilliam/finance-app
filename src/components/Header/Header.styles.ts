import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleHeader = StyleSheet.create({
  container: {
    width: `${theme.withScreen.big}%`,
    borderBottomWidth: 2, 
    paddingBottom: 10,
    marginBottom: 20,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: "center",
    borderBottomColor: theme.light.border,
    backgroundColor: theme.light.backgroundPrimary, 

  },
  icon: {
    marginRight: 10,
    color: theme.light.icon,
    fontSize: theme.size.iconMedium,
  },
  title: {
    fontSize: theme.size.fontMedium,
    fontWeight: 'bold',
    justifyContent: 'flex-start'
  }
});

export default StyleHeader;