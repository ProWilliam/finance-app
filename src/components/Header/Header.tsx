import React from 'react';
import { View, Text } from 'react-native';

// Component
import IconBank from '../../assets/icons/IconBank';

// Styles
import styles from './Header.styles';


const Header = () => {
  return(
    <View style={styles.container}>
      <IconBank style={styles.icon} />
      <Text style={styles.title}>
        Banco
      </Text>
    </View>
  )
}
export default Header;