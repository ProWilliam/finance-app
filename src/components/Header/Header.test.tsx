import React from 'react';
import { Text, View } from 'react-native';
import renderer from 'react-test-renderer';

// Component adn Styles
import IconBank from '../../assets/icons/IconBank';
import Header from './Header';
import styles from './Header.styles';

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders the title correctly', () => {
    const tree = renderer.create(<Header />);
    const textComponent = tree.root.findByType(Text);
    expect(textComponent.props.children).toBe('Banco');
  });

  it('should render the IconBank component', () => {
    const tree = renderer.create(<Header />);
    const iconBank = tree.root.findByType(IconBank);
    expect(iconBank).toBeTruthy();
  });

  it('should apply the correct styles to the container', () => {
    const tree = renderer.create(<Header />);
    const containerStyle = tree.root.findByType(View).props.style;
    expect(containerStyle).toEqual(styles.container);
  });

});