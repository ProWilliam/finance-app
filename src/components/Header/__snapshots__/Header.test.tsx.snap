// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#eaeaea",
      "borderBottomWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
      "paddingBottom": 10,
      "width": "100%",
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    aria-hidden="true"
    bbHeight={50}
    bbWidth={50}
    fill="none"
    focusable={false}
    height={50}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#000000",
          "fontSize": 35,
          "marginRight": 10,
        },
        {
          "flex": 0,
          "height": 50,
          "width": 50,
        },
      ]
    }
    tintColor="#000000"
    vbHeight={24}
    vbWidth={24}
    width={50}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      font={
        {
          "fontSize": 35,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M8 7V6a1 1 0 0 1 1-1h11a1 1 0 0 1 1 1v7a1 1 0 0 1-1 1h-1M3 18v-7a1 1 0 0 1 1-1h11a1 1 0 0 1 1 1v7a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1Zm8-3.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "fontSize": 34,
        "fontWeight": "bold",
        "justifyContent": "flex-start",
      }
    }
  >
    Banco
  </Text>
</View>
`;
